### PR TITLE
`ultralytics 8.3.111` YOLOv10 skip `one2many` head when fused

### LINKED ISSUE #20191: The number of parameters and GFLOPs for yolov10 are different from those stated on the official website
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I got a different result when i using the latest version of ultralytics. especially in yolov10, as the figure1 show, i got a different layers, parameters, gradients and GFLOPs.

Image
but the setting of those things in  doc/model_data.py like the figure 2:
Image

### Additional

_No response_

### PR DESCRIPTION
**Pull request by Y-T-G**
Closes #20191 

Not required if not training.

**main**
```python
In [1]: from ultralytics import YOLO

In [2]: model = YOLO("yolov10n.pt")

In [3]: model.fuse()
YOLOv10n summary (fused): 125 layers, 2,762,608 parameters, 0 gradients, 8.6 GFLOPs
```

**PR**
```python
In [6]: from ultralytics import YOLO

In [7]: model = YOLO("yolov10n.pt")

In [8]: model.fuse()
YOLOv10n summary (fused): 102 layers, 2,299,264 parameters, 0 gradients, 6.7 GFLOPs
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This update improves the YOLO11 model by optimizing its architecture and updating documentation outputs, resulting in a cleaner, more efficient model and clearer example results. 🚀📈

### 📊 Key Changes
- **Version Bump:** Upgraded Ultralytics package version from 8.3.110 to 8.3.111.
- **Model Optimization:** Added a `fuse` method to the YOLO11 detection head to remove the one2many head, simplifying the model for deployment.
- **Fusing in Tasks:** Updated the model fusion process to call the new `fuse` method for YOLO11, ensuring the one2many head is removed during model export or optimization.
- **Tutorial Output Updates:** Refreshed outputs in the example tutorial notebook to reflect the latest version, environment, and performance metrics.

### 🎯 Purpose & Impact
- **Simpler, Leaner Models:** Removing the one2many head during fusion makes exported YOLO11 models smaller and more efficient, which is especially beneficial for deployment and inference. ⚡
- **Improved Documentation:** Updated tutorial outputs provide users with the most current and accurate information, reducing confusion and making it easier to follow along. 📚
- **Better User Experience:** These changes help ensure that users get optimized models and up-to-date guidance, leading to smoother workflows and improved results. 😊

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To help ensure a smooth integration of your improvement, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and follow the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is up to date with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address those issues.
- ✅ **Update Documentation**: Update the [documentation](https://docs.ultralytics.com/) for any new or modified features as needed.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm all tests pass.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to only what is necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For further details, check out our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). If you have any questions, please leave a comment below. Thank you for your valuable contribution to Ultralytics! 🚀🔥

_(This is an automated response; an Ultralytics engineer will review your PR and assist you soon.)_

**Comment by glenn-jocher** [comment]
Nice, not sure why it didn't do this in the previous PR though (turns out one is an alias of the other)
https://github.com/ultralytics/ultralytics/pull/20193/commits/097542784aa31829db91f893e61d8554776dbba9

**Comment by Laughing-q** [comment]
@Y-T-G Nice! I was wondering if this is going to affect the training started from yolov10 pretrained weights as it eliminates two pretrained layers, then I realized that we are not using `fuse` here when loading the model, hence no worries about that at all.
https://github.com/ultralytics/ultralytics/blob/ee2b03ee2f97cb16277153c9f4c65329c8670d65/ultralytics/engine/model.py#L290-L290

**Comment by UltralyticsAssistant** [comment]
🎉 Fantastic work, @Y-T-G, @Laughing-q, and @glenn-jocher! Merging this PR marks a significant step forward for YOLO11, making the model cleaner, faster, and easier to use for everyone in the Ultralytics community. As Henry Ford said, "If everyone is moving forward together, then success takes care of itself." Your collaborative efforts—optimizing the architecture, streamlining deployment, and updating documentation—will empower users with leaner models and clearer guidance. Thank you for your dedication and for making a lasting impact! 🚀
